src/extension/background-script/actions/nostr/generatePrivateKey.ts

```diff
@@ -1,54 +1,41 @@
-import * as secp256k1 from "@noble/secp256k1";
-import Hex from "crypto-js/enc-hex";
-import sha512 from "crypto-js/sha512";
-import type { MessagePrivateKeyGenerate } from "~/types";
+import { encryptData } from "~/common/lib/crypto";
+import type { MessageNostrPrivateKeyGenerate } from "~/types";
 
 import state from "../../state";
 
-const generatePrivateKey = async (message: MessagePrivateKeyGenerate) => {
-  const type = message?.args?.type;
-
-  const privateKey =
-    type === "random" ? generateRandomKey() : await deriveKey();
+const generatePrivateKey = async (message: MessageNostrPrivateKeyGenerate) => {
+  const id = message.args?.id || state.getState().currentAccountId;
+  const password = await state.getState().password();
+  if (!password) {
+    return {
+      error: "Password is missing.",
+    };
+  }
+  const accounts = state.getState().accounts;
+
+  if (id && Object.keys(accounts).includes(id)) {
+    const mnemonic = await state.getState().getMnemonic();
```

Comment:
`generatePrivateKey` supports passing an id arg but we always use the mnemonic of the currently-selected account. I think this could cause unexpected behaviour. Thoughts?

Reply:
I changed this to create a new mnemonic object for the current account